### PR TITLE
Update tests to verify Component Graph support in pipelines

### LINKED ISSUE #2168: Pipeline: support ComponentGraph instance as input
In #2091 our pipelines now require a `component_graph` parameter. This parameter can either be a) a list of components, either by string name for our core set or a component class, or b) a dictionary describing the component graph in the format required by `ComponentGraph`

I think we should support a third option: providing a `ComponentGraph` instance directly to the pipeline init. It should be a small change :)
https://github.com/alteryx/evalml/pull/2091/files#diff-e3cc5bc8079883f5d6d5d526a2e769765d139a49d1c6528dbb0f6ccd83aa75d2R72

### PR DESCRIPTION
**Pull request by bchen1116**
fix #2168

### CODE REVIEW

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 1]
```diff
@@ -2796,3 +2796,76 @@ def test_training_only_component_in_pipeline_transform(X_y_binary):
...
+    )
+
+    component_graph3 = ComponentGraph(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Undersampler": ["Undersampler", "Imputer.x", "y"],
+        }
+    )
+
+    BinaryClassificationPipeline(component_graph1)
```
Comment (freddyaboulton) [inline evalml/tests/pipeline_tests/test_pipelines.py:2832]: Can we add `BinaryClassificationPipeline(component_graph1).component_graph == componentgraph1` to make sure everything is wired correctly under the hood?

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @bchen1116 ! Thank you.

File: evalml/pipelines/pipeline_base.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ class PipelineBase(ABC, metaclass=PipelineBaseMeta):
     """Machine learning pipeline.
 
     Args:
-        component_graph (list or dict): List of components in order. Accepts strings or ComponentBase subclasses in the list.
+        component_graph (ComponentGraph, list, dict): ComponentGraph instance, list of components in order, or dictionary of components.
```
[the remark this answers is not in the document]
Comment (bchen1116) [reply evalml/pipelines/pipeline_base.py:46]: Good catch! Will do

File: evalml/tests/pipeline_tests/test_pipelines.py [hunk 3]
```diff
@@ -2796,3 +2796,82 @@ def test_training_only_component_in_pipeline_transform(X_y_binary):
     pipeline.fit(X, y)
     transformed = pipeline.transform(X)
     assert len(transformed) == len(X) - 2
+
+
+def test_component_graph_pipeline():
+    component_graph1 = ComponentGraph(
```
Comment (angela97lin) [inline evalml/tests/pipeline_tests/test_pipelines.py:2802]: Just curious, what are each of these three trying to test? Could be useful to name the variables accordingly :)

**Review by angela97lin — APPROVED** [review]
LGTM, thank you! 😁

File: evalml/tests/pipeline_tests/test_pipelines.py (hunk 3 above)
Comment (bchen1116) [reply evalml/tests/pipeline_tests/test_pipelines.py:2802]: Mm, these are just checking that we can fit the regression, classification, and no estimator component graphs into the pipelines. Can change the name to make it more clear!

**Review by chukarsten — APPROVED** [review]
This looks good to me.

**Pull request merged**